File: core/src/main/java/com/JavaBruse/core/sniffer/domain/DTO/ConnectionInsightDTO.java
package com.JavaBruse.core.sniffer.domain.DTO;

import lombok.Data;

import java.util.List;

@Data
public class ConnectionInsightDTO {
    private String localIp;
    private List<Integer> localPorts;
    private String remoteIp;
    private Integer remotePort;
    private Long totalPackets;
    private Long totalBytes;
    private Long firstPacketTime;
    private Long lastPacketTime;
    private Long synCount;
    private Long finCount;
    private Long rstCount;
    private Long identifiedPackets;
    private List<JA4CandidateDTO> ja4Candidates;
    private List<SNICandidateDTO> sniCandidates;

    @Data
    public static class JA4CandidateDTO {
        private String id;
        private String fingerprint;
        private String application;
        private String device;
        private String os;
        private Long count;
        private Integer confidence;
        private Integer hop;
    }

    @Data
    public static class SNICandidateDTO {
        private String id;
        private String sni;
        private String service;
        private Long count;
        private Integer confidence;
        private Integer hop;
    }
}

File: frontend/src/app/handler-package.component/connection-insight.DTO.ts
export interface ConnectionInsight {
    localIp: string;
    localPorts: number[];
    remoteIp: string;
    remotePort: number;
    totalPackets: number;
    totalBytes: number;
    firstPacketTime: number;
    lastPacketTime: number;
    synCount: number;
    finCount: number;
    rstCount: number;
    identifiedPackets: number;
    ja4Candidates: JA4Candidate[];
    sniCandidates: SNICandidate[];
}

export interface JA4Candidate {
    id: string;
    fingerprint: string;
    application: string;
    device: string;
    os: string;
    count: number;
    confidence: number;
    hop: number;
}

export interface SNICandidate {
    id: string;
    sni: string;
    service: string;
    count: number;
    confidence: number;
    hop: number;
}